zenpy/lib/request.py

```diff
@@ -204,20 +204,41 @@ class UploadRequest(RequestHandler):
     """ Handles uploading files to Zendesk. """
 
     def post(self, fp, token=None, target_name=None, content_type=None, api_object=None):
+        file_pointer_properties = self._get_file_pointer_properties(fp, target_name=target_name)
```

Comment:
This is all very sensitive code and I was hoping we could avoid touching it for this macro functionality. I think either this code is refactored properly with tests for all the many cases it has to handle or we leave it alone entirely. Less likely to break things this way.

Reply:
Also should mention you don't necessarily have to handle every type of file-like object/path like the existing code does.